lib/Graph/CustomOpUtils.cpp

```diff
@@ -0,0 +1,232 @@
+/**
+ * Copyright (c) Glow Contributors. See CONTRIBUTORS file.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glow/Graph/CustomOpUtils.h"
+
+namespace glow {
+
+// Maps CustomOpDataType to glow ElemKind.
+static ElemKind customOpDataTypeToElemKind(CustomOpDataType dtype) {
+  ElemKind elemK;
+  switch (dtype) {
+  case DTFloat32:
+    elemK = ElemKind::FloatTy;
+    break;
+  case DTFloat16:
+    elemK = ElemKind::Float16Ty;
+    break;
+  case DTQInt8:
+    elemK = ElemKind::Int8QTy;
+    break;
+  case DTQUInt8:
+    elemK = ElemKind::UInt8QTy;
+    break;
+  case DTIInt32:
+    elemK = ElemKind::Int32ITy;
+    break;
+  case DTIInt64:
+    elemK = ElemKind::Int64ITy;
+    break;
+  default:
+    llvm_unreachable("Type not supported yet");
+  }
+  return elemK;
+}
+
+CustomOpIOTensor glowTypeToCustomOpIOTensor(const TypeRef type) {
+  CustomOpIOTensor iot;
+
+  // Copy dims.
+  auto dims = type->dims();
+  iot.rank = dims.size();
+  iot.dims = (int32_t *)malloc(iot.rank * sizeof(int32_t));
+
+  for (int i = 0; i < iot.rank; i++)
+    iot.dims[i] = dims[i];
+
+  switch (type->getElementType()) {
+  case ElemKind::FloatTy:
+    iot.dtype = CustomOpDataType::DTFloat32;
+    break;
+  case ElemKind::Float16Ty:
+    iot.dtype = CustomOpDataType::DTFloat16;
+    break;
+  case ElemKind::Int8QTy:
+    iot.dtype = DTQInt8;
+    break;
+  case ElemKind::UInt8QTy:
+    iot.dtype = DTQUInt8;
+    break;
+  case ElemKind::Int32ITy:
+    iot.dtype = CustomOpDataType::DTIInt32;
+    break;
+  case ElemKind::Int64ITy:
+    iot.dtype = CustomOpDataType::DTIInt64;
+    break;
+  default:
+    llvm_unreachable("Type not supported yet");
+  }
+
+  // Explicitly set to nullptr when only holding type information.
+  iot.data = nullptr;
+
+  return iot;
+}
+
+CustomOpIOTensor glowTensorToCustomOpIOTensor(Tensor *tensor) {
+  CustomOpIOTensor iot = glowTypeToCustomOpIOTensor(&tensor->getType());
+  iot.data = tensor->getUnsafePtr();
+  return iot;
+}
+
+CustomOpIOTensor initializeCustomOpIOTensor(const int32_t &rank,
```

Comment:
nit: For primitive types like ints we can just pass by value. 
```suggestion
CustomOpIOTensor initializeCustomOpIOTensor(int32_t rank,
```